src/main/scala/thurloe/dataaccess/HttpSendGridDAO.scala

```diff
@@ -32,18 +32,35 @@ class HttpSendGridDAO extends SendGridDAO with LazyLogging {
     }
   }
 
-  def lookupPreferredEmail(userId: RawlsUserSubjectId): Future[RawlsUserEmail] =
-    for {
-      email <- dataAccess
-        .lookup(userId.value, "contactEmail")
-        .recoverWith { _ =>
-          val fallbackEmail = dataAccess.lookup(userId.value, "email")
-          logger.info(
-            s"Failed to get stored contactEmail for ${userId.value}. Defaulting to stored email ($fallbackEmail) for user."
-          )
-          fallbackEmail
+  //Looks up a KVP, converting empty values or missing KVPs into None
+  private def lookupNonEmptyKeyValuePair(userId: String, key: String) = {
+    dataAccess.lookup(userId, key).map { kvp =>
+      if(kvp.keyValuePair.value.isEmpty) None
+      else Some(kvp)
+    }.recover { _ => None }
+  }
+
+  //There are two cases that need to be handled when looking up the preferred contact email:
+  // 1) If the contactEmail is not present at all, the DB query will throw an exception. So that needs to be handled.
+  // 2) If the contactEmail is present but blank, it also needs to be ignored. Thurloe accepts arbitrary key/value pairs
+  //    and makes no guarantees about what the data might look like, so a blank value or invalid email is a valid case.
+  //    Orchestration does it's best to enforce valid emails, but blank emails are still used when a user removes their
+  //    contact email (instead of deleting the k/v pair entirely). There is a larger discussion to be had about the future
+  //    of this service and how to structure the data more reliably, but for now we'll deal with the current realities and
+  //    defend against them here.
+  //  ^-- note: the same can also all be said about the 'email' k/v pair, but a blank email is a less common case due to
+  //      how profiles are populated.
+  def lookupPreferredEmail(userId: RawlsUserSubjectId): Future[RawlsUserEmail] = {
+    lookupNonEmptyKeyValuePair(userId.value, "contactEmail") flatMap {
+      case Some(kvp) => Future.successful(RawlsUserEmail(kvp.keyValuePair.value)) //contactEmail was found and non-empty
+      case None =>
+        logger.info(s"Failed to get stored contactEmail for ${userId.value}. Defaulting to account email for user.")
+        lookupNonEmptyKeyValuePair(userId.value, "email") flatMap {
+          case Some(kvp) => Future.successful(RawlsUserEmail(kvp.keyValuePair.value)) //account email was found and non-empty
+          case None => Future.failed(new KeyNotFoundException(userId.value, "email"))
```

Comment:
this is one of the most important parts: throwing this KeyNotFoundException will cause the monitor to ack the PubSub message and prevent a backlog from forming